File: ArcherApp/app/src/main/java/me/mxcsyounes/archernotebook/viewmodels/AdjustmentViewModel.java
package me.mxcsyounes.archernotebook.viewmodels;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;



import java.util.List;

import me.mxcsyounes.archernotebook.repositories.AdjustmentsRepository;
import me.mxcsyounes.archernotebook.database.entities.Adjustment;

public class AdjustmentViewModel extends AndroidViewModel {

    private LiveData<List<Adjustment>> mAllAdjustment;
    private AdjustmentsRepository mRepository;

    public AdjustmentViewModel(@NonNull Application application) {
        super(application);
        mRepository = new AdjustmentsRepository(application);
        mAllAdjustment = mRepository.getAllAdj();
    }

    public LiveData<List<Adjustment>> getAllAdjustments() {
        return mAllAdjustment;
    }

    public void insertAdjustment(Adjustment adjustment) {
        mRepository.insertAdjustment(adjustment);
    }

    public void deleteAdjustment(Adjustment adjustment) {
        mRepository.deleteAdjustment(adjustment);
    }

    public void deleteAllAdjustments() {
        mRepository.deleteAllAdjustments();
    }
}

File: ArcherApp/app/src/main/java/me/mxcsyounes/archernotebook/viewmodels/AdjustmentViewModel.kt
package me.mxcsyounes.archernotebook.viewmodels;


import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData
import me.mxcsyounes.archernotebook.database.entities.Adjustment
import me.mxcsyounes.archernotebook.repositories.AdjustmentsRepository

class AdjustmentViewModel(application: Application) : AndroidViewModel(application) {

    val mAllAdjustment: LiveData<MutableList<Adjustment>>
    private val mRepository: AdjustmentsRepository = AdjustmentsRepository(application)

    init {
        mAllAdjustment = mRepository.allAdj
    }


    fun insertAdjustment(adjustment: Adjustment) =
            mRepository.insertAdjustment(adjustment);


    fun deleteAdjustment(adjustment: Adjustment) =
            mRepository.deleteAdjustment(adjustment);


    fun deleteAllAdjustments() =
            mRepository.deleteAllAdjustments();

}
